File: Data Structures & Algorithms/lowest-common-ancestor-in-binary-search-tree/submission-9.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def lowestCommonAncestor(self, root: TreeNode, p: TreeNode, q: TreeNode) -> TreeNode:
        if not root:
            return root

        i = root

        while True:
            if max(p.val, q.val) < i.val:
                i = i.left
            elif min(p.val, q.val) > i.val:
                i = i.right
            else:
                return i

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-9.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {number[][]}
     */
    levelOrder(root) {
        const q = []
        const result = []

        if (root) {
            q.push(root)
        }

        while (q.length) {
            const n = q.length
            const currLevel = []
            for (let i = 0; i < n; i++) {
                const node = q.shift()
                currLevel.push(node.val)

                if (node.left) {
                    q.push(node.left)
                }
                if (node.right) {
                    q.push(node.right)
                }
            }
            result.push(currLevel)
        }
        return result
    }
}
